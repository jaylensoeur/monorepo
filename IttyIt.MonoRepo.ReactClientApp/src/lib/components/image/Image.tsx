import * as React from 'react';
import { useState } from 'react';

type Props = {
    imageUrl: string
    alt?: string
    title?: string
    className? : string
    render?: (data: {isLoading: boolean}) => void
};

const Image = ({render, className, imageUrl, alt, title}: Props): JSX.Element => {
    const [isLoading, setIsLoading] = useState(true);

    const handleImageLoaded = () => {
        setIsLoading(false);
    };

    return (
        <>
            <img data-testid="image"
                className={className}
                alt={alt}
                title={title}
                src={imageUrl}
                onLoad={handleImageLoaded}
            />
            {render ? render({isLoading}) : ''}
        </>
    );
};

export default Image;
