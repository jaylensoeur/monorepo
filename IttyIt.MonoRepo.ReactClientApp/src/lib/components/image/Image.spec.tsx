import * as React from "react";
import { create } from "react-test-renderer";

import Image from "./Image";

describe("<Image />", () => {
    it("should render", async () => {
        const src = "test.jpg";
        const component = create(<Image imageUrl={src}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("should accept a src", async () => {
        const src = "test.jpg";
        const component = create(<Image imageUrl={src}/>);
        const instance = component.root;
        const img = instance.findByType("img");

        expect(img.props.src).toBe(src);
    });

    it("should render a child component", async () => {
        const src = "test.jpg";
        const postsCallbackMock = jest.fn();

        create(<Image imageUrl={src} render={postsCallbackMock}/>);
        expect(postsCallbackMock).toHaveBeenCalled();
    });
});
