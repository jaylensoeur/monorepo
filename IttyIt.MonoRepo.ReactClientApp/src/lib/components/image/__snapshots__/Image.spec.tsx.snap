// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render 1`] = `
Array [
  <img
    data-testid="image"
    onLoad={[Function]}
    src="test.jpg"
  />,
  "",
]
`;
